dev/com.ibm.ws.security.token.ltpa/src/com/ibm/ws/security/token/ltpa/internal/LTPAConfigurationImpl.java

```diff
@@ -156,18 +158,34 @@ private void loadConfig(Map<String, Object> props) {
         // expirationDifferenceAllowed is set to 3 seconds (3000ms) by default.
         // If expirationDifferenceAllowed is set to less than 0, then the two expiration values will not be compared in the LTPAToken2.decrypt() method.
         expirationDifferenceAllowed = (Long) props.get(KEY_EXP_DIFF_ALLOWED);
-        monitorDirectory = (Boolean) props.get(CFG_KEY_MONITOR_DIRECTORY);
-
-        resolveActualPrimaryKeysFileLocation();
+        enableDirectoryMonitoring = (Boolean) props.get(CFG_KEY_ENABLE_DIRECTORY_MONITORING);
+        updateTrigger = (String) props.get(CFG_KEY_UPDATE_TRIGGER);
 
         //get all validationKeys elements
         Map<String, List<Map<String, Object>>> validationKeysElements = Nester.nest(props, CFG_KEY_VALIDATION_KEYS);
         if (!validationKeysElements.isEmpty()) {
+            isValidationKeysFileConfigured = !validationKeysElements.get(CFG_KEY_VALIDATION_KEYS).isEmpty();
+        }
+
+        resolveActualPrimaryKeysFileLocation();
```

Comment:
If you move the resolveActualPrimaryKeysFileLocation to this line, then in previous method, the primary key may not have a fully qualify path name

Reply:
The primary key value has already been resolved up to this point. This call is only used to resolve the directory, which is why it is fine if we do it after the nest call.